### PR TITLE
Feature videoconverter windows

### PR DESCRIPTION
**Pull request by danielaSantaCruz**

### CODE REVIEW

File: src/main/java/org/fundacion/jala/converter/ConverterApplication.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /**
  * Copyright (c) 2021 Fundacion Jala.
- *
+ * <p>
```
Comment (Paola-Xaq) [inline src/main/java/org/fundacion/jala/converter/ConverterApplication.java]: delete, maybe an error when writing. Is in all the classes, review it.

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+public class Converter {
+    private String startFirstCommand = "ffmpeg -i ";
+    private VideoParameter parameter;
+    private String format;
+    private String output;
+    private String pathOutput;
+    private static final int WAIT_TIME = 5000;
+    private static final int INIT_NUMBER = 20;
+
+    public Converter(final VideoParameter vParameter) {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java]: The name of the parameter should be more descriptive

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+     * @param newMetaData boolean that indicates if the video has to have metadata file
+     */
+    public void setMetaData(final boolean newMetaData) {
+        this.metaData = newMetaData;
+    }
+
+    /**
+     * Gets the name of the input video
+     */
+    public String getInputPath() {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java:101]: You need to specify that it returns

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        return inputPath;
+    }
+
+    /**
+     * Gets the name of the output video
+     */
+    public String getOutputFormat() {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java:109]: You need to specify that it returns

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+        return outputFormat;
+    }
+
+    /**
+     * Gets the resolution of the output video
+     */
+    public String getResolution() {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java:117]: You need to specify that it returns

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+        return resolution;
+    }
+
+    /**
+     * Gets the width of the output video
+     */
+    public int getWidth() {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java:125]: You need to specify that it returns

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+        return width;
+    }
+
+    /**
+     * Gets the height of the output video
+     */
+    public int getHeight() {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java:133]: You need to specify that it returns

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+        return height;
+    }
+
+    /**
+     * Gets the audio of the output video
+     */
+    public boolean hasAudio() {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java:141]: You need to specify that it returns

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+        return audio;
+    }
+
+    /**
+     * Gets the frame rate of the output video
+     */
+    public int getFrameRate() {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java:149]: You need to specify that it returns

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+        return frameRate;
+    }
+
+    /**
+     * Gets the tumbnail of the output video
+     */
+    public boolean hasTumbnail() {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java:157]: You need to specify that it returns

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+        return tumbnail;
+    }
+
+    /**
+     * Gets the metadata of the output video
+     */
+    public boolean hasMetaData() {
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java:165]: You need to specify that it returns

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java (hunk 2 above)
Comment (cristianfchq) [reply src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java]: I agree with you Edson

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java [hunk 12]
```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) 2021 Fundacion Jala.
+ * <p>
```
Comment (edsonarios) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/VideoParameter.java]: This line seems to be too much

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java [hunk 13]
```diff
@@ -0,0 +1,120 @@
...
+     * Changes the resolution and aspect ratio of the input video
+     * @return resolution command
+     */
+    private String changeResolution() {
+        int width = parameter.getWidth();
+        int height = parameter.getHeight();
+        if (width > 0 && height > 0) {
+            String scale = width + ":" + height;
+            String acpectRatio = ":force_original_aspect_ratio=decrease,pad=";
+            String resolutionCommand = "-vf \"scale=" + scale + acpectRatio + scale + ":-1:-1:color=white\"";
```
Comment (cristianfchq) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java]: I think the variable declaration should be at the beginning of the method.

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java [hunk 14]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    /**
+     * Generates a input video tumbnail
+     * @return tumbnail command
+     */
+    private String generateATumbnail() {
+        boolean tumbnail = parameter.hasTumbnail();
+        if (tumbnail) {
+            String tumbnailCommand = " -ss 00:00:01 -vframes 1 " + pathOutput + "VideoTumbnail.png";
```
Comment (cristianfchq) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java]: I think the variable declaration should be at the beginning of the method.

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java [hunk 15]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    /**
+     * Changes the input video frame rate
+     * @return frame command
+     */
+    private String changeFrameRate() {
+        int frameRate = parameter.getFrameRate();
+        if (frameRate > INIT_NUMBER) {
+            String frameCommand = " -r " + frameRate + " -y ";
```
Comment (cristianfchq) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java]: I think the variable declaration should be at the beginning of the method.

**Review by Paola-Xaq — APPROVED** [review]
JavaDocs needs some params, some in return, review it.
And there is an extra line in the Copyright Section.

Code and logic seems good, nice work. :)

File: src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java [hunk 16]
```diff
@@ -0,0 +1,120 @@
...
+    }
+
+    /**
+     * Removes the audio of the input video
+     * @return audio command
+     */
+    private String removeAudio() {
+        boolean audio = parameter.hasAudio();
+        if (audio) {
+            String audioCommand = "-an ";
```
Comment (cristianfchq) [inline src/main/java/org/fundacion/jala/converter/service/videoclasses/Converter.java]: I think the variable declaration should be at the beginning of the method.

File: src/main/java/org/fundacion/jala/converter/ConverterApplication.java (hunk 1 above)
Comment (danielaSantaCruz) [reply src/main/java/org/fundacion/jala/converter/ConverterApplication.java]: It is fixed :)

**Review by cristianfchq — APPROVED** [review]

**Review by edsonarios — APPROVED** [review]

**Pull request merged**